ipatests/test_xmlrpc/tracker/certmapdata.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright (C) 2015  FreeIPA Contributors see COPYING for license
+#
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from nose.tools import assert_raises
+
+from ipalib.errors import MutuallyExclusiveError, RequirementError
+from ipapython.dn import DN
+from ipatests.util import assert_deepequal
+
+
+class CertmapdataMixin(object):
+    certmapdata_options = {u'issuer', u'subject', u'usercertificate',
+                           u'ipacertmapdata'}
+
+    def _data_from_options(self, **options):
+        issuer = None
+        subject = None
+
+        def any_in(keys, mapping):
+            return any([(key in mapping) for key in keys])
+
+        if not any_in(self.certmapdata_options, options):
+            raise RequirementError(name=u'certmapdata')
+
+        if (
+            any_in((u'ipacertmapdata', u'usercertificate',), options) and
+            any_in((u'issuer', u'subject',), options)
+        ):
+            raise MutuallyExclusiveError(reason=u'Mutually exclusive options '
+                                                u'provided at the same time.')
+
```

Comment:
You can also add a requirement: if subject or issuer is provided, then the other one is also required.